src/lib.rs

```diff
@@ -442,3 +442,24 @@ storage_types! {
         }
     }
 }
+
+/// Shared tools for parsing quantities from strings.
+pub mod str {
+    /// Represents an error encountered while parsing a string into a `Quantity`.
+    #[derive(Clone, Copy, Debug, PartialEq)]
```

Comment:
Should this drop `Copy` and add `Eq`? Standard library error types use `#[derive(Debug, Clone, PartialEq, Eq)]`. Not sure why `Copy` is excluded. Perhaps only to make copying an error more explicit?